File: Transpiled/Core/Semantic/Symbol.cpp
#include "Symbol.hpp"

namespace NumbatLogic
{
	class Symbol;
}
#line 0 "/home/cliffya/git/Lang/Source/Core/Semantic/Symbol.nll"
namespace NumbatLogic
{
#line 3 "/home/cliffya/git/Lang/Source/Core/Semantic/Symbol.nll"
#line 33 "/home/cliffya/git/Lang/Source/Core/Semantic/Symbol.nll"
	Symbol::Symbol()
	{
		m_eKind = Kind::NAMESPACE;
		m_sShortName = 0;
		m_sQualifiedName = 0;
		m_pDeclAST = 0;
		m_pScope = 0;
#line 35 "/home/cliffya/git/Lang/Source/Core/Semantic/Symbol.nll"
		m_eKind = Kind::VAR;
		m_sShortName = "";
		m_sQualifiedName = "";
		m_pDeclAST = 0;
		m_pScope = 0;
	}

}

File: Transpiled/Core/Semantic/Symbol.cs
#line 1 "/home/cliffya/git/Lang/Source/Core/Semantic/Symbol.nll"
namespace NumbatLogic
{
	class Symbol
	{
		public enum Kind
		{
			NAMESPACE,
			CLASS,
			ENUM,
			GENERIC_PARAM,
			FUNCTION,
			DELEGATE,
			METHOD,
			VAR,
			PARAM,
			ENUM_VALUE,
		}

		public Kind m_eKind;
#line 22 "/home/cliffya/git/Lang/Source/Core/Semantic/Symbol.nll"
		public string m_sShortName;
#line 25 "/home/cliffya/git/Lang/Source/Core/Semantic/Symbol.nll"
		public string m_sQualifiedName;
#line 28 "/home/cliffya/git/Lang/Source/Core/Semantic/Symbol.nll"
		public AST m_pDeclAST;
#line 31 "/home/cliffya/git/Lang/Source/Core/Semantic/Symbol.nll"
		public SymbolScope m_pScope;
#line 33 "/home/cliffya/git/Lang/Source/Core/Semantic/Symbol.nll"
		public Symbol()
		{
			m_eKind = Kind.VAR;
			m_sShortName = "";
			m_sQualifiedName = "";
			m_pDeclAST = null;
			m_pScope = null;
		}

	}
}

